bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/discovery/HomekitMdnsDiscoveryParticipant.java

```diff
@@ -0,0 +1,167 @@
+            Map<String, String> properties = getProperties(service);
+
+            String macAddress = properties.get("id"); // MAC address
+            String ipAddress = service.getHostAddresses()[0]; // ipV4 address
+            int port = service.getPort();
+            if (port != 0) {
+                ipAddress = ipAddress + ":" + port;
+            }
+
+            AccessoryCategory category;
+            try {
+                String ci = properties.getOrDefault("ci", ""); // accessory category
+                category = AccessoryCategory.from(Integer.parseInt(ci));
+            } catch (IllegalArgumentException e) {
+                category = null;
+            }
+
+            if (ipAddress != null && macAddress != null && category != null) {
+                DiscoveryResultBuilder builder = DiscoveryResultBuilder.create(uid);
+                builder.withLabel(THING_LABEL_FMT.formatted(service.getName(), macAddress)) //
+                        .withProperty(CONFIG_HOST_NAME, getHostName(service)) //
+                        .withProperty(CONFIG_IP_ADDRESS, ipAddress) //
+                        .withProperty(Thing.PROPERTY_MAC_ADDRESS, macAddress) //
+                        .withProperty(PROPERTY_ACCESSORY_CATEGORY, category.toString()) //
+                        .withRepresentationProperty(Thing.PROPERTY_MAC_ADDRESS);
+
+                if (properties.get("md") instanceof String model) {
+                    builder.withProperty(Thing.PROPERTY_MODEL_ID, model);
+                }
+                if (properties.get("s#") instanceof String serial) {
+                    builder.withProperty(Thing.PROPERTY_SERIAL_NUMBER, serial);
+                }
+                if (properties.get("pv") instanceof String protocolVersion) {
+                    builder.withProperty(PROPERTY_PROTOCOL_VERSION, protocolVersion);
+                }
+
+                return builder.build();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo service) {
+        Map<String, String> properties = getProperties(service);
+
+        String mac = properties.get("id"); // MAC address
+        AccessoryCategory cat;
+        try {
+            String ci = properties.getOrDefault("ci", "");
+            cat = AccessoryCategory.from(Integer.parseInt(ci));
+        } catch (IllegalArgumentException e) {
+            cat = null;
+        }
+
+        if (mac != null && cat != null) {
+            return new ThingUID(AccessoryCategory.BRIDGE == cat ? THING_TYPE_BRIDGE : THING_TYPE_ACCESSORY,
+                    mac.replace(":", "").toLowerCase()); // thing id example "a1b2c3d4e5f6"
+        }
+
+        return null;
+    }
+
+    /**
+     * The JmDNS library getProperties() method has a bug whereby it fails to return any properties
+     * in the case that the TXT record contains zero length parts. This is a drop in replacement.
```

Comment:
It would be good to mention this as a maintainer note, so that this can be tracked and the work-around eventually be removed when jmdns 3.6.3 is released and included into openHAB:
```suggestion
     * in the case that the TXT record contains zero length parts. This is a drop in replacement.
     * Fixed upstream by https://github.com/jmdns/jmdns/pull/355
```